### PR TITLE
Support for OCaml 4.13.0

### PR DESCRIPTION
**Pull request by cphalen**
Between OCaml 4.10.0 (previous max supported version) and OCaml 4.13.0 there were two major changes:

- `Stream` was moved out of the stdlib into an opam package called `camlp-streams`, so I added the opam package to parts of the codebase that required `Stream`
- `Parsetree` was updated which required some changes:
      - Support for `Pwith_modtype` and `Pwith_modtypesubst` variants, these with constraints allow you to specify module type and module type substitutions
      - Support for `Psig_modtypesubst` variant which allows signatures to be defined with module type substitutions
      - Rearranging some type deconstruction for `Ppat_construct` which was restructured slightly

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -13,10 +13,8 @@ jobs:
           - macos-latest
           - ubuntu-latest
         ocaml-version:
-          - 4.10.0
-          - 4.10.1
-          - 4.11.0
-          - 4.11.1
+          - 4.13.0
+          - 4.13.1
```
Comment (cphalen) [inline .github/workflows/workflow.yml:17]: Not so sure that deprecating support for 4.10.* and 4.11.* is what we want but I can't think of any way we can continue to support the older versions if we want to also support the newer version

Comment (KeenWill) [reply .github/workflows/workflow.yml:17]: Yup that's fine the ASTs are incompatible so we have to deprecate the old versions.

**Comment by Vighnesh-V** [comment]
lgtm chief ;)

**Review by KeenWill — CHANGES_REQUESTED** [review]
Code changes look good! Can you update `CHANGELOG.md` with the changes you made? Maybe bump version to 2.0.0 and also put that version into the opam and dune files?

**Comment by cphalen** [comment]
> Code changes look good! Can you update `CHANGELOG.md` with the changes you made? Maybe bump version to 2.0.0 and also put that version into the opam and dune files?

Oh great point I'll do this today thanks Will

**Review by KeenWill — APPROVED** [review]
looks good i'll make a PR to opam later today

**Pull request merged**